src/adagrams.js

```diff
@@ -1,15 +1,140 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+
+function getRandomProperty() {
+  const keys = Object.keys(LETTER_POOL);
+
+  return keys[Math.floor(Math.random() * keys.length)];
+}
+
+let drawnLetters = [];
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  while (drawnLetters.length < 10) {
+    let myCurrentKey = getRandomProperty();
+    let myCurrentValue = LETTER_POOL[myCurrentKey];
+    if (myCurrentValue !== 0) {
+      drawnLetters.push(myCurrentKey);
+      LETTER_POOL[myCurrentKey] = myCurrentValue - 1;
+    }
+    if (LETTER_POOL[myCurrentKey] === 0) {
+      delete LETTER_POOL[myCurrentKey];
+    }
+  }
+  
+  return drawnLetters;
 };
 
+
+
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+  let bankCopy = JSON.parse(JSON.stringify(lettersInHand))
```

Comment:
This is a good way to copy a list. One thing to think about however is that the list we are copying simply contains letters, not objects, so a shallow copy using the spread operator will work fine here!

     bankCopy = [...lettersInHand];